File: tests/fixtures/go/service.go
package main

import (
	"fmt"

	"example.com/fixture/store"
)

// DefaultRegion is a package-level variable.
var DefaultRegion = "eu"

// Build returns an engine for the region.
func Build(region string) store.Engine {
	engine := store.Engine{Region: region}
	engine.Ping()
	return engine
}

// Inspect takes an interface value and calls through it.
func Inspect(item store.Labeled) string {
	return item.Describe()
}

func main() {
	fmt.Println(Inspect(Build(DefaultRegion)))
}

File: tests/fixtures/php/src/Service.php
<?php

declare(strict_types=1);

namespace Fixture;

use Fixture\Models\Base;
use Fixture\Models\Engine;
use Fixture\Models\Region;
use GuzzleHttp\Client;

function build(Region $region): Engine
{
    $engine = new Engine($region);
    $engine->ping();

    return $engine;
}

function inspect(Base $item): string
{
    return $item->describe();
}

function fetchOne(int $ident): string
{
    $client = new Client();
    $response = $client->get("/engines/{$ident}");

    return (string) $response->getBody();
}

function defaultRegion(): Region
{
    return Region::from('eu');
}
